jupyter_client/kernelspec.py

```diff
@@ -73,6 +73,21 @@ def __init__(self, name):
         self.name = name
 
 class KernelSpecManager(LoggingConfigurable):
+
+    kernel_spec_class = DottedObjectName(
```

Comment:
Yeah, we realized that Type was simpler after we had started using DON + factory in various places. We haven't updated everywhere yet, but we are at least using Type in new places. Looks like you got it worked out.